### DISCUSSION TITLE
Access the row instance in the onRowSelectionChange callback function

### DISCUSSION THREAD

**Discussion by elliotgonzalez123** [TanStack/table#4810] [Ideas]
I'd like to propose adding a feature where we can get access to the selected `row` instance in the `onRowSelectionChange` callback.  Currently, the callback is an updater function that sets the row selection state using and id and a boolean `{ 1 : true}`.  It would be incredibly useful if the selected row instance is also available in this callback.  This would allow us to add the `row.original` to another piece of state being managed in the component during the `onChange` event.  

ex:

```ts
 onRowSelectionChange: (fn, row) => {
      setRowSelection(fn);
      setSelectedRows(row.original)
    },
```

This will come in handy when creating tables where selecting a row immediately shows or aggregates data in the UI somewhere.  The current solution is to do this in a `useEffect`:

```ts
 React.useEffect(() => {
    if (rowSelection) {
      const rows = table
        .getSelectedRowModel()
        .flatRows.map((row) => row.original);
      setSelectedRows(rows);
    }
  }, [setSelectedRows, rowSelection, table]);
```
It would be great to be able to remove this `useEffect` and just do it in the callback.  Here is an old discussion on the issue: https://github.com/TanStack/table/discussions/2155

**Comment by Jdruwe** [comment]
I assume this is still an 'issue'?

**Comment by transon-top** [comment]
If this feature were implemented, that's great

**Comment by JoorgeSv** [comment]
This works for me.
First your data must have an id.
`const yourData = [ {id: 123, ...}, {id: 456, ...}, {id: 789, ...}, ...]`
then
```jsx
const [selection, setSelection] = useState({ state: {}, items: [] });
const table = useReactTable({
    columns,
    data: yourData,
    getCoreRowModel: getCoreRowModel(), 
    enableMultiRowSelection: true,
    getRowId: row => row.id,
    state: { rowSelection: selection.state },
    onRowSelectionChange: getNewState => {
      const newState = getNewState(selection.state);
      const selectedItems = yourData.filter(item => newState[item.id]);
      //you can add any other logic you require with the newState or selectedItems
      setSelection({ state: newState, items: selectedItems })
    }, 
});

console.log("selected items:", selection.items);
```
